### DISCUSSION TITLE
save as an image via contextmenu not work

### DISCUSSION THREAD

**Discussion by Wangxin666888** [MicrosoftEdge/WebView2Feedback#4563] [Q&A] [answered]
I use webview2 to display a page, and there is an img on this page.
`<img src="https://abc.com/image?id=8"/>`
I add a web resource filter to response this image, 
`webview.CoreWebView2.WebResourceRequested += (s, e) => configuration.HandleWebResourceRequested(e);` 

when i right click on the image to show a contextmenu, and then click "Save as" to open save dialog. all is ok but nothing has been saved.

why? and how to solve it?

**Comment by victorhuangwq** [comment]
@yildirimcagri-msft do you have a clue on this?

**Reply by Wangxin666888** [reply]
Thanks!!!

when i changed the response with 206 code to 200 code, it worked.

**Comment by ghost** [comment]
> I use webview2 to display a page, and there is an img on this page.
> 
> `<img src="https://abc.com/image?id=8"/>`
> 
> I add a web resource filter to response this image, 
> 
> `webview.CoreWebView2.WebResourceRequested += (s, e) => configuration.HandleWebResourceRequested(e);` 
> 
> 
> 
> when i right click on the image to show a contextmenu, and then click "Save as" to open save dialog. all is ok but nothing has been saved.
> 
> 
> 
> 
> 
> why? and how to solve it?  

Hi, does it work if you don't have the WebResourceRequested event handler?

**Reply by Wangxin666888** [reply]
Thanks!!!

when i changed the response with 206 code to 200 code, it worked.

**Comment by Wangxin666888** [comment, marked as answer]
when i changed the response with 206 code to 200 code, it worked.
